### PR TITLE
[macOS][NativeWindowing] Modernize input code, fix XBMC shortcuts

### PR DESCRIPTION
**Pull request by enen92**
## Description
Unfortunately the input code used in the native windowing implementation makes use of deprecated frameworks like Carbon and rely on CoreGraphics when more modern alternatives are available. The code to detect the keypress is a bit ugly, all we need is available as part of the NSEvent itself (plus the constants provided by AppKit).

https://github.com/xbmc/xbmc/commit/36e8cbafcc3a530e3de105dd89ec5a6cf79540db -> removes the global application shortcuts (like globe+f, cmd+ctrl+f, etc). Since we moved the input handling to the `NSView` (as an `NSResponder`) these are all handled by the window itself (as a result of registering them with the `NSMenu`. At this point this code is not called, making it effectively dead code. The previous move also fixed an issue with the inconsistency of state in the app menu "toggle float on top". I've only kept those that shortcuts that are not mapped by the application (like screenshot and hide). Removing CApp dependency is a great side-effect too IMHO :)

https://github.com/xbmc/xbmc/commit/e97523ef9b58cd337fb5fd252be0c47736930acd -> modernizes the input code, dropping Carbon.h and CoreGraphics. No functional changes. 

https://github.com/xbmc/xbmc/commit/992f049b7930df7ef6126f52885f5f3e520cdcff -> actually fixes the kodi shortcuts like ctrl+shift+d by finding the actual key without modifiers.

[f28353a873f82c2c4a88dd4bafa3bff392d1809d ](https://github.com/xbmc/xbmc/pull/23081/commits/f28353a873f82c2c4a88dd4bafa3bff392d1809d) -> implements the application hide shortcut properly (we are using minimize currently which is duplicated)

### CODE REVIEW

**Comment by enen92** [comment]
jenkins build this please

**Comment by enen92** [comment]
Added a new commit to implement the Hide Menu item (we are currently calling minimize which is wrong)

**Review by fuzzard — APPROVED** [review]
Lets do it. Changes look reasonable, only thing that came out at me was the rename of the NSString to unicode (from unicodeNSString). Just seems a little less clear of what it is but its such a minor thing itreally doesnt matter either way, so go with what you have.

File: xbmc/platform/darwin/osx/XBMCApplication.mm [hunk 1]
```diff
@@ -97,11 +104,28 @@ - (void)applicationWillFinishLaunching:(NSNotification*)notification
                                                        keyEquivalent:@"t"];
   floatOnTopMenuItem.keyEquivalentModifierMask = NSEventModifierFlagCommand;
   [windowMenu addItem:floatOnTopMenuItem];
+
+  NSMenuItem* hideMenuItem = [[NSMenuItem alloc] initWithTitle:@"Hide"
+                                                        action:@selector(hideAppAction:)
+                                                 keyEquivalent:@"h"];
+  hideMenuItem.keyEquivalentModifierMask = NSEventModifierFlagCommand;
+  [windowMenu addItem:hideMenuItem];
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/osx/XBMCApplication.mm:112]: normally Hide is under the \<app name\> menu (the first one, `appMenu` in code I guess)

File: xbmc/platform/darwin/osx/XBMCApplication.mm [hunk 2]
```diff
@@ -97,11 +104,28 @@ - (void)applicationWillFinishLaunching:(NSNotification*)notification
...
+
   [windowMenu addItemWithTitle:@"Minimize"
                         action:@selector(performMiniaturize:)
                  keyEquivalent:@"m"];
 }
 
+- (BOOL)validateMenuItem:(NSMenuItem*)item
+{
+  // validate if the hide menu item should be shown
+  if ([item.title isEqual:@"Hide"] && [[NSApplication sharedApplication] isHidden])
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/osx/XBMCApplication.mm:122]: standard behavior is to have Show in place of Hide :)

**Review by kambala-decapitator — APPROVED** [review]

File: xbmc/platform/darwin/osx/XBMCApplication.mm (hunk 2 above)
Comment (enen92) [reply xbmc/platform/darwin/osx/XBMCApplication.mm:121]: I've addressed both on a new commit. Just checked hide/show usually is shown in app submenu!
Thanks!

**Comment by enen92** [comment]
@kambala-decapitator good to go due to the last commit?

**Review by kambala-decapitator — APPROVED** [review]
I started checking about a minute before your comment :)

**Pull request merged**